### PR TITLE
[PIL-964][AV] - Repayments Contact Name and Email

### PR DESCRIPTION
**Pull request by aashik-varahan**

### CODE REVIEW

File: app/forms/RepaymentsContactEmailFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import javax.inject.Inject
+
+class RepaymentsContactEmailFormProvider @Inject() extends Mappings {
+  def apply(contactName: String): Form[String] =
+    Form(
+      "contactEmail" -> text("repayments.contactEmail.error.contactEmail.required", Seq(contactName))
+        .verifying(
+          firstError(
+            maxLength(Constants.MAX_LENGTH_100, "repayments.contactEmail.error.contactEmail.length"),
```
Comment (jassalrichy) [inline app/forms/RepaymentsContactEmailFormProvider.scala:31]: Should be using MAX_LENGTH_132 for emails

File: app/pages/PaymentRefundAmountPage.scala
Comment (jassalrichy) [inline app/pages/PaymentRefundAmountPage.scala:1]: Add unit tests.

File: app/pages/RepaymentsContactEmailPage.scala
Comment (jassalrichy) [inline app/pages/RepaymentsContactEmailPage.scala:1]: Add unit tests.

File: app/pages/RepaymentsContactNamePage.scala
Comment (jassalrichy) [inline app/pages/RepaymentsContactNamePage.scala:1]: Add unit tests.

File: app/views/repayments/RepaymentsContactEmailView.scala.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+@import viewmodels.InputWidth._
+@import config.FrontendAppConfig
+@import viewmodels.LabelSize.Large
+@import views.html.components.gds.sectionHeader
+
+@this(
+    layout: templates.Layout,
+    formHelper: FormWithCSRF,
+    sectionHeader: sectionHeader,
```
Comment (jassalrichy) [inline app/views/repayments/RepaymentsContactEmailView.scala.html:25]: remove unused sectionHeader.

File: app/views/repayments/RepaymentsContactNameView.scala.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+@import viewmodels.InputWidth._
+@import config.FrontendAppConfig
+@import viewmodels.LabelSize.Large
+@import views.html.components.gds.sectionHeader
+
+@this(
+    layout: templates.Layout,
+    formHelper: FormWithCSRF,
+    sectionHeader: sectionHeader,
```
Comment (jassalrichy) [inline app/views/repayments/RepaymentsContactNameView.scala.html:25]: Remove unused sectionHeader.

File: test/controllers/repayments/RepaymentsContactNameControllerSpec.scala
Comment (jassalrichy) [inline test/controllers/repayments/RepaymentsContactNameControllerSpec.scala:1]: NonUkBankControllerSpec is a required spec. Why has this been renamed in this branch ?

File: test/navigation/RepaymentNavigatorSpec.scala
Comment (jassalrichy) [inline test/navigation/RepaymentNavigatorSpec.scala:1]: Navigation spec needs updating for RepaymentsContactEmailPage & RepaymentsContactNamePage

File: app/forms/RepaymentsContactNameFormProvider.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import play.api.data.Form
+import javax.inject.Inject
+
+class RepaymentsContactNameFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "contactName" -> text("repayments.contactName.error.required")
+        .verifying(
+          firstError(
```
[the remark this answers is not in the document]
Comment (aashik-varahan) [reply app/forms/RepaymentsContactNameFormProvider.scala]: Removed firstError condition

File: conf/messages.en [hunk 5]
```diff
@@ -1229,6 +1264,19 @@ repayments.nonUKBank.error.iban.length = IBAN must be up to 34 characters
 repayments.nonUKBank.error.bic.format = Enter a valid BIC or SWIFT code like HBUKGB4B
 repayments.nonUKBank.error.iban.format = Enter a valid IBAN like GB29NWBK60161331926819
 
+repayments.contactName.title = What is the name of the person or team we should contact about the refund request?
+repayments.contactName.heading = What is the name of the person or team we should contact about the refund request?
+repayments.contactName.hint = For example, ‘Tax team’ or ‘Ashley Smith’.
+repayments.contactName.error.required = Enter name of the person or team we should contact for this refund request
+repayments.contactName.error.length = Name of the contact person or team must be 100 characters or less
+
+repayments.contactEmail.title = What is the email address? New
```
[the remark this answers is not in the document]
Comment (aashik-varahan) [reply conf/messages.en]: Updated the content.

File: app/forms/RepaymentsContactEmailFormProvider.scala (hunk 1 above)
Comment (aashik-varahan) [reply app/forms/RepaymentsContactEmailFormProvider.scala:31]: As discussed, the new requirement is to use 100 characters for Emails as mentioned on the ticket

File: app/pages/RepaymentsContactEmailPage.scala
Comment (aashik-varahan) [reply app/pages/RepaymentsContactEmailPage.scala:1]: Added

File: app/pages/RepaymentsContactNamePage.scala
Comment (aashik-varahan) [reply app/pages/RepaymentsContactNamePage.scala:1]: Added

File: app/views/repayments/RepaymentsContactEmailView.scala.html (hunk 2 above)
Comment (aashik-varahan) [reply app/views/repayments/RepaymentsContactEmailView.scala.html]: Removed sectionHeader

File: app/views/repayments/RepaymentsContactNameView.scala.html (hunk 3 above)
Comment (aashik-varahan) [reply app/views/repayments/RepaymentsContactNameView.scala.html]: Removed sectionHeader

File: test/controllers/repayments/RepaymentsContactNameControllerSpec.scala
Comment (aashik-varahan) [reply test/controllers/repayments/RepaymentsContactNameControllerSpec.scala:1]: Added back NonUKBankControllerSpec

File: test/navigation/RepaymentNavigatorSpec.scala
Comment (aashik-varahan) [reply test/navigation/RepaymentNavigatorSpec.scala:1]: Updated with new unit tests

**Review by javed-iqbal1 — APPROVED** [review]

File: app/pages/RepaymentsContactEmailPage.scala
Comment (jassalrichy) [inline app/pages/RepaymentsContactEmailPage.scala:1]: Missing spec for this view.

File: app/views/repayments/RepaymentsContactEmailView.scala.html
Comment (jassalrichy) [inline app/views/repayments/RepaymentsContactEmailView.scala.html:1]: Missing spec for this view.

File: app/views/repayments/RepaymentsContactNameView.scala.html
Comment (jassalrichy) [inline app/views/repayments/RepaymentsContactNameView.scala.html:1]: Missing spec for this view.

File: app/views/repayments/RepaymentsContactEmailView.scala.html
Comment (aashik-varahan) [reply app/views/repayments/RepaymentsContactEmailView.scala.html:1]: Added Spec for Repayments Contact Email Form and View

File: test/views/repayments/RepaymentsContactEmailViewSpec.scala
Comment (jassalrichy) [inline test/views/repayments/RepaymentsContactEmailViewSpec.scala:1]: Can we add tests for error content please.

File: test/views/repayments/RepaymentsContactNameViewSpec.scala
Comment (jassalrichy) [inline test/views/repayments/RepaymentsContactNameViewSpec.scala:1]: Can we add tests for error content please.

File: test/views/repayments/RepaymentsContactEmailViewSpec.scala
Comment (aashik-varahan) [reply test/views/repayments/RepaymentsContactEmailViewSpec.scala:1]: Added now

File: test/views/repayments/RepaymentsContactNameViewSpec.scala
Comment (aashik-varahan) [reply test/views/repayments/RepaymentsContactNameViewSpec.scala:1]: Added now

File: test/views/repayments/RepaymentsContactEmailViewSpec.scala [hunk 6]
```diff
@@ -18,36 +18,117 @@ package views.repayments
...
     }
 
-    "have a hint" in {
-      view.getElementsByClass("govuk-hint").text must include(
-        "We will only use this to contact you about " +
-          "this refund request."
-      )
+    "value entered exceeds character limit" should {
+
+      val contactEmail = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA@gmail.com"
```
Comment (jassalrichy) [inline test/views/repayments/RepaymentsContactEmailViewSpec.scala:83]: Could replace with:
val contactEmail = "".padTo(100, 'A').concat("gmail.com")

File: test/views/repayments/RepaymentsContactNameViewSpec.scala [hunk 7]
```diff
@@ -18,39 +18,86 @@ package views.repayments
...
+        view.getElementsByClass("govuk-error-message").text must include("Enter name of the person or team we should contact for this refund request")
+      }
+
     }
 
-    "have a button" in {
-      view.getElementsByClass("govuk-button").text must include("Continue")
+    "value entered exceeds character limit" should {
+
+      val contactName = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
```
Comment (jassalrichy) [inline test/views/repayments/RepaymentsContactNameViewSpec.scala:82]: Could replace with:
val contactEmail = "".padTo(100, 'A').concat("gmail.com")

File: test/views/repayments/RepaymentsContactEmailViewSpec.scala (hunk 6 above)
Comment (aashik-varahan) [reply test/views/repayments/RepaymentsContactEmailViewSpec.scala]: Done

File: test/views/repayments/RepaymentsContactNameViewSpec.scala (hunk 7 above)
Comment (aashik-varahan) [reply test/views/repayments/RepaymentsContactNameViewSpec.scala]: Done

**Review by jassalrichy — APPROVED** [review]

**Review by javed-iqbal1 — APPROVED** [review]

**Pull request merged**